zephyr/esp_shared/components/driver/rtc_io.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * SPDX-FileCopyrightText: 2019-2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "esp_err.h"
+#include "esp_check.h"
+#include "driver/rtc_io.h"
+#include "hal/rtc_io_hal.h"
+
+static const char __attribute__((__unused__)) *RTCIO_TAG = "RTCIO";
+
+struct {
+	unsigned int spinlock;
+} rtcio_context = {
+	0
+};
+
+#define RTCIO_ENTER_CRITICAL() do { rtcio_context.spinlock = irq_lock(); } while(0)
```

Comment:
Can we get rid of any `_ENTER_CRITICAL` and `EXIT_CRITICAL` from all files?  
I think the all macros in functions can be replace to `rtcio_context.spinlock = irq_lock();`  instead.